pkg/ibmvpc/parser.go

```diff
@@ -14,9 +14,96 @@ import (
 	vpcmodel "github.com/np-guard/vpc-network-config-analyzer/pkg/vpcmodel"
 )
 
+//nolint:funlen,gocyclo //styles are too long and can not be split
+func FilterByResourceGroup(rc *datamodel.ResourcesContainerModel, resourceGroup string) {
```

Comment:
I see two problems with this function:
1. It repeats the same code pattern many times
2. If we add a new resource to the collector's data model, we might forget to update this function as well.

I have two possible alternatives:
1. Implement in the collector's data model a function similar to the one here (only solves the second issue)
2. Postpone the filtering a little bit, and implement in `VPCConfigsFromResources` by extending the current `shouldSkipByVPC` mechanism. The nice thing about this alternative is that it can be used to filter by region as well, and that it doesn't duplicate code that much.

Please experiment with these alternatives, and see if they make things any better.

Reply:
@adisos , do you have any better idea?

Reply:
> I agree this mechanism can be used and extended for this as well, and can save those similar code patterns. But if one runs the analyzer with collection as well (instead of input vpc-config file), and asks to dump the config object into a file, will it be with the filtered resources or not?

Filtering is also applied by the collector in its API calls. This already works.
So, yes - resources will be filtered